File: IBM-bank-backend/src/main/java/com/IBM/IBM_bank/Models/Movimentacao.java
package com.IBM.IBM_bank.Models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import java.time.LocalDate;

@Entity
public class Movimentacao {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "conta_id", nullable = false)
    @JsonIgnore
    private Conta conta;

    @ManyToOne
    @JoinColumn(name = "conta_credito_id")
    @JsonIgnore
    private ContaCredito contaCredito;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private StatusPagamento statusPagamento; // "PENDENTE" ou "PAGO"

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TipoMovimentacao tipo; // "DEBITO" ou "CREDITO"

    @Column(nullable = false, length = 10)
    private Double valor;
    private String descricao;

    @Column(nullable = false)
    private LocalDate dataMovimentacao;

    private LocalDate dataCompetencia; // Mês e ano da competência (e.g., "MM-yyyy")

    // Getters e Setters

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Conta getConta() {
        return conta;
    }

    public void setConta(Conta conta) {
        this.conta = conta;
    }

    public ContaCredito getContaCredito() {
        return contaCredito;
    }

    public void setContaCredito(ContaCredito contaCredito) {
        this.contaCredito = contaCredito;
    }

    public TipoMovimentacao getTipo() {
        return tipo;
    }

    public void setTipo(TipoMovimentacao tipo) {
        this.tipo = tipo;
    }

    public Double getValor() {
        return valor;
    }

    public void setValor(Double valor) {
        this.valor = valor;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    public StatusPagamento getStatusPagamento() {
        return statusPagamento;
    }

    public void setStatusPagamento(StatusPagamento statusPagamento) {
        this.statusPagamento = statusPagamento;
    }

    public LocalDate getDataMovimentacao() {
        return dataMovimentacao;
    }

    public void setDataMovimentacao(LocalDate dataMovimentacao) {
        this.dataMovimentacao = dataMovimentacao;
    }

    public LocalDate getDataCompetencia() {
        return dataCompetencia;
    }

    public void setDataCompetencia(LocalDate dataCompetencia) {
        this.dataCompetencia = dataCompetencia;
    }
}

File: IBM-bank-frontend/src/shared/models/movimentacao.ts
export enum TipoMovimentacao {
    DEBITO = 'DEBITO',
    CREDITO = 'CREDITO'
}

export enum StatusPagamento {
    PENDENTE = 'PENDENTE',
    PAGO = 'PAGO'
}
export interface Movimentacao {
    id: number;
    contaId?: number; 
    contaCreditoId?: number; 
    tipo: TipoMovimentacao; 
    valor: number; 
    dataMovimentacao: string; 
    descricao: string; 
    statusPagamento?: StatusPagamento; 
    dataCompetencia?: string; 
}

export enum eColunaXPropMovimentacao {
    // 'Nome da coluna' = 'nome da propriedade'
    'ID' = 'id',
    'Data da Movimentação' = 'dataMovimentacao',
    'Tipo' = 'tipo',
    'Valor' = 'valor',
    'Status do Pagamento' = 'statusPagamento',
    'Data de Competência' = 'dataCompetencia',
    'Conta ID' = 'contaId',
    'Conta Crédito ID' = 'contaCreditoId',
  }